### PR TITLE
Adds explanatory prose related to `delegate`.

### PR DESCRIPTION
**Pull request by ioannad**
This resolves the last three so far unaddressed `todo::` sections:
- Add prose explaining try-delegate's jump.
- Adds prose and intuition for delegating exception handlers.
  + Also a minor modification to the prose of catching exception handlers, that somewhat reduces calling the administrative instruction by its identifier (CAUGHTadm) in the prose.

### CODE REVIEW

File: document/core/syntax/instructions.rst [hunk 1]
```diff
@@ -686,8 +686,8 @@ In case of |BLOCK| or |IF| it is a *forward jump*,
 resuming execution after the matching |END|.
 In case of |LOOP| it is a *backward jump* to the beginning of the loop.
 
-.. todo::
-   Add prose for try-delegate's jump.
+In case of |TRY|--|DELEGATE| catching an exception, it is a *forward jump*,
+throwing the caught exception *before* the matching |END|.
```
Comment (rossberg) [inline document/core/syntax/instructions.rst]:
```suggestion
When |TRY|--|DELEGATE| handles an exception, it also behaves similar to a forward jump,
effectively rethrowing the caught exception right before the matching |END|.
```

File: document/core/exec/runtime.rst [hunk 2]
```diff
@@ -542,21 +542,20 @@ to their associated branch *targets*, each of which is expressed syntactically a
 :ref:`instructions <syntax-instr>` possibly following a :ref:`tag address <syntax-tagaddr>`.
 If there is no :ref:`tag address <syntax-tagaddr>`, the instructions of that handler clause correspond to a |CATCHALL| clause.
 
-.. todo::
-   Add prose for delegating handlers.
+Delegating handlers start with the identifier |DELEGATEadm| and carry a label index,
+indicating the outer block in which any caught exceptions will be thrown (delegated to).
```
Comment (rossberg) [inline document/core/exec/runtime.rst]: See my email suggestion regarding a slight refactor of this syntax.

Comment (ioannad) [reply document/core/exec/runtime.rst]: I wanted to change this prose here to match #226, but on second thought it doesn't make sense without the rest of the changed text. So I am reverting these changes and adding this missing prose to #226 instead.

Comment (ioannad) [reply document/core/exec/runtime.rst]: Done.

**Review by takikawa — APPROVED** [review]

**Pull request merged**